src/cursor.rs

```diff
@@ -0,0 +1,116 @@
+use std::ops::{Add, AddAssign, Index, IndexMut, Sub, SubAssign};
+
+// TODO: AddAssign
+pub struct CursorMut<'a, T> {
+    data: &'a mut [T],
+    index: usize,
+}
+
+impl<'a, T> CursorMut<'a, T> {
+    pub fn new(data: &'a mut [T]) -> Self {
```

Comment:
Should we have a `From` `impl` for this conversion?

Reply:
Sure, seems reasonable. Done.